### PR TITLE
feat: Add tests for QR Scan Presenter

### LINKED ISSUE #139: Write tests for QR Scan Activity
Currently, tests for QR Scan Activity are missing. 

I'm working on it

### PR DESCRIPTION
**Pull request by iamareebjamal**
- Add tests for unit functions, data flow and view integrations
- Fix failing tests by checks and appropriate calls
- Change contract to make QR Presenter more testable

Fixes #139

### CODE REVIEW

**Comment by iamareebjamal** [comment]
@mejariamol Please review

File: app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRPresenter.java [hunk 1]
```diff
@@ -50,17 +48,31 @@ private void processBarcode(String barcode) {
...
+                scanQRView.onScannedAttendee(attendee);
+            });
+    }
+
+    public void setAttendees(List<Attendee> attendees) {
+        this.attendees = attendees;
     }
 
     @Override
     public void attach() {
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRPresenter.java]: `attach` method should be opposite of `detach`. That is if view is released in `detach` then it should be reassigned in `attach`.
Please make these changes in other presenters too.

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRPresenter.java]: You are correct and we don't have this behaviour correctly, but how is this related to this PR?

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRPresenter.java]: Please create a new issue for that

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRPresenter.java]: Yeah it's not related to this PR.
Actually I found this while reading `shouldNotAccessViewAfterDetach()`. where presenter is detached first and then attached again. Then view's accessibility is tested. 

I will open new issue for this. :+1:

**Comment by iamareebjamal** [comment]
@niranjan94 Please merge
